internal/datastore/postgres/postgres_test.go

```diff
@@ -738,6 +757,239 @@ func ConcurrentRevisionHeadTest(t *testing.T, ds datastore.Datastore) {
+				seenWatchRevisionsLock.Unlock()
+
+				time.Sleep(1 * time.Millisecond)
+			case <-withCancel.Done():
+				return
+			}
+		}
+	}()
+
+	<-waitForWatch
+
+	// Write the two concurrent transactions, while watching for changes.
+	g := errgroup.Group{}
+
+	waitToStart := make(chan struct{})
+	waitToFinish := make(chan struct{})
+
+	var commitLastRev, commitFirstRev datastore.Revision
+	g.Go(func() error {
+		var err error
+		commitLastRev, err = ds.ReadWriteTx(ctx, func(rwt datastore.ReadWriteTransaction) error {
+			err = rwt.WriteRelationships(ctx, []*core.RelationTupleUpdate{
+				tuple.Touch(tuple.MustParse("something:001#viewer@user:123")),
+				tuple.Touch(tuple.MustParse("something:002#viewer@user:123")),
+				tuple.Touch(tuple.MustParse("something:003#viewer@user:123")),
+			})
+			require.NoError(err)
+
+			close(waitToStart)
+			<-waitToFinish
+
+			return err
+		})
+		require.NoError(err)
+		return nil
+	})
+
+	<-waitToStart
+
+	commitFirstRev, err = ds.ReadWriteTx(ctx, func(rwt datastore.ReadWriteTransaction) error {
+		return rwt.WriteRelationships(ctx, []*core.RelationTupleUpdate{
+			tuple.Touch(tuple.MustParse("resource:1001#reader@user:456")),
+			tuple.Touch(tuple.MustParse("resource:1002#reader@user:456")),
+			tuple.Touch(tuple.MustParse("resource:1003#reader@user:456")),
+		})
+	})
+	close(waitToFinish)
+
+	require.NoError(err)
+	require.NoError(g.Wait())
+
+	// Ensure the revisions do not compare.
+	require.False(commitFirstRev.GreaterThan(commitLastRev))
+	require.False(commitLastRev.GreaterThan(commitFirstRev))
+	require.False(commitFirstRev.Equal(commitLastRev))
+
+	// Write another revision.
+	afterRev, err := ds.ReadWriteTx(ctx, func(rwt datastore.ReadWriteTransaction) error {
+		rtu := tuple.Touch(&core.RelationTuple{
+			ResourceAndRelation: &core.ObjectAndRelation{
+				Namespace: "resource",
+				ObjectId:  "2345",
+				Relation:  "reader",
+			},
+			Subject: &core.ObjectAndRelation{
+				Namespace: "user",
+				ObjectId:  "456",
+				Relation:  "...",
+			},
+		})
+		return rwt.WriteRelationships(ctx, []*core.RelationTupleUpdate{rtu})
+	})
+	require.NoError(err)
+	require.True(afterRev.GreaterThan(commitFirstRev))
+	require.True(afterRev.GreaterThan(commitLastRev))
+
+	// Ensure that the last revision is eventually seen from the watch.
+	require.Eventually(func() bool {
+		seenWatchRevisionsLock.Lock()
+		defer seenWatchRevisionsLock.Unlock()
+		return len(seenWatchRevisions) == 3 && seenWatchRevisions[len(seenWatchRevisions)-1].String() == afterRev.String()
+	}, 2*time.Second, 5*time.Millisecond)
+}
+
+func OverlappingRevisionWatchTest(t *testing.T, ds datastore.Datastore) {
+	require := require.New(t)
+
+	ctx := context.Background()
+	r, err := ds.ReadyState(ctx)
+	require.NoError(err)
+	require.True(r.IsReady)
+
+	rev, err := ds.HeadRevision(ctx)
+	require.NoError(err)
+
+	pds := ds.(*pgDatastore)
+	require.True(pds.watchEnabled)
+
+	prev := rev.(postgresRevision)
+	nextx := prev.snapshot.xmax + 1
```

Comment:
super nit, feel free to ignore: 
```suggestion
	nexttx := prev.snapshot.xmax + 1
```